universal-application-tool-0.0.1/app/auth/UATProfile.java

```diff
@@ -1,11 +1,46 @@
 package auth;
 
+import com.google.common.base.Preconditions;
+import java.time.Clock;
+import models.Account;
+import models.Applicant;
 import org.pac4j.core.profile.CommonProfile;
 
 public class UATProfile extends CommonProfile {
+  private final Clock clock;
+  private Applicant applicant;
+
   public UATProfile() {
+    this(Clock.systemDefaultZone());
+  }
+
+  public UATProfile(Clock clock) {
     super();
-    // TODO: write this to the database, read the ID back, and use that as the ID here.
-    setId("TODO");
+    this.clock = clock;
+  }
+
+  public Applicant getApplicant() {
+    if (applicant == null) {
+      this.applicant = new Applicant();
+      this.applicant.id = Long.valueOf(this.getId());
+      this.applicant.refresh();
+    }
+    return applicant;
+  }
+
+  public Account getAccount() {
+    return getApplicant().getAccount();
+  }
```

Comment:
Thought (non-blocking): it seems odd that it loads the account through the applicant, I'd have expected vice versa

Reply:
Hm, yeah, I see what you mean - I'll swap that.  It's easier to fetch the "one" from the "many", but nbd.